drivers/SmartThings/zigbee-sensor/src/init.lua

```diff
@@ -0,0 +1,157 @@
+-- Copyright 2025 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local ZigbeeDriver = require "st.zigbee"
+local defaults = require "st.zigbee.defaults"
+local clusters = require "st.zigbee.zcl.clusters"
+local IASZone = clusters.IASZone
+local capabilities = require "st.capabilities"
+local ZONETYPE = "ZoneType"
+local constants = require "st.zigbee.constants"
+local PowerConfiguration = clusters.PowerConfiguration
+local device_management = require "st.zigbee.device_management"
+
+local CONTACT_SWITCH = 0x0015
+local MOTION_SENSOR = 0x000D
+local WATER_SENSOR = 0x002A
+
+local ZIGBEE_GENERIC_SENSOR_PROFILE = "generic-sensor"
+local ZIGBEE_GENERIC_CONTACT_SENSOR_PROFILE = "generic-contact-sensor"
+local ZIGBEE_GENERIC_MOTION_SENSOR_PROFILE = "generic-motion-sensor"
+local ZIGBEE_GENERIC_WATERLEAK_SENSOR_PROFILE = "generic-waterleak-sensor"
+
+-- ask device to upload its zone type
+local ias_device_added = function(driver, device)
+  device:send(IASZone.attributes.ZoneType:read(device))
+end
+
+-- ask device to upload its zone status, then the status of capabilities can be synchronized
+local ias_info_changed = function(driver, device)
```

Comment:
This function will be run every time the device is updated on the cloud side.  You should not be reading an attribute here.  If the intent is for this to happen at some point after the device is added, you should use the `call_with_delay` to have the driver wait some period and then send the read.